packages/expo-cli/src/commands/build-native/easJson.ts

```diff
@@ -0,0 +1,182 @@
+export enum Workflow {
+  Generic = 'generic',
+  Managed = 'managed',
+}
+
+export interface AndroidManagedPreset {
+  workflow: Workflow.Managed;
+  buildType?: 'app-bundle' | 'apk';
+}
+
+export interface AndroidGenericPreset {
+  workflow: Workflow.Generic;
+  buildCommand?: string;
+  artifactPath?: string;
+  withoutCredentials?: boolean;
+}
+
+export interface iOSManagedPreset {
+  workflow: Workflow.Managed;
+  buildType?: 'archive' | 'simulator';
+}
+
+export interface iOSGenericPreset {
+  workflow: Workflow.Generic;
+}
+
+export type AndroidPreset = AndroidManagedPreset | AndroidGenericPreset;
+export type iOSPreset = iOSManagedPreset | iOSGenericPreset;
+
+interface EasJson {
+  builds: {
+    credentialsSource: CredentialsSource;
+    android?: { [key: string]: AndroidManagedPreset | AndroidGenericPreset };
+    ios?: { [key: string]: iOSManagedPreset | iOSGenericPreset };
+  };
+}
+
+// EasConfig represents eas.json with one specific preset
+export interface EasConfig {
+  builds: {
+    credentialsSource: CredentialsSource;
+    android?: AndroidManagedPreset | AndroidGenericPreset;
+    ios?: iOSManagedPreset | iOSGenericPreset;
+  };
+}
+
+const EasJsonSchema = Joi.object({
+  builds: Joi.object({
+    credentialsSource: Joi.string().valid('local', 'remote', 'auto').default('auto'),
+    android: Joi.object().pattern(
+      Joi.string(),
+      Joi.object({
+        workflow: Joi.string().valid('generic', 'managed').required(),
+      }).unknown(true)
+    ),
+    ios: Joi.object().pattern(
+      Joi.string(),
+      Joi.object({
+        workflow: Joi.string().valid('generic', 'managed').required(),
+      }).unknown(true)
+    ),
+  }),
+});
+
+const AndroidGenericSchema = Joi.object({
+  workflow: Joi.string().valid('generic').required(),
+  buildCommand: Joi.string(),
+  artifactPath: Joi.string(),
+  withoutCredentials: Joi.boolean(),
+});
+
+const AndroidManagedSchema = Joi.object({
+  workflow: Joi.string().valid('managed').required(),
+  buildType: Joi.string().valid('apk', 'app-bundle'),
+});
+
+const iOSGenericSchema = Joi.object({
+  workflow: Joi.string().valid('generic').required(),
+});
+const iOSManagedSchema = Joi.object({
+  workflow: Joi.string().valid('managed').required(),
+  buildType: Joi.string().valid('archive', 'simulator'),
+});
+
+const schemaPresetMap: Record<string, Record<string, Joi.Schema>> = {
+  android: {
+    generic: AndroidGenericSchema,
+    managed: AndroidManagedSchema,
+  },
+  ios: {
+    managed: iOSManagedSchema,
+    generic: iOSGenericSchema,
+  },
+};
+
+interface Options {
+  platform: 'android' | 'ios' | 'all';
+  credentialsSource?: CredentialsSource;
+}
+
+export class EasJsonReader {
+  constructor(private projectDir: string, private options: Options) {}
+
+  public async read(presetName: string): Promise<EasConfig> {
+    const easJson = await this.readFile();
+
+    let androidConfig;
+    if ([Platform.Android, 'all'].includes(this.options.platform)) {
```

Comment:
```suggestion
    if (['android', 'all'].includes(this.options.platform)) {
```
Please be consistent. Use either `Platform.Android` or `android` but consequently.